apis/Google.Storage.V1/Google.Storage.V1/SignedUrlUtils.cs

```diff
@@ -0,0 +1,410 @@
+        /// <param name="requestMethod">The HTTP request method for which the signed URL is allowed to be used. May be null,
+        /// in which case GET will be used.</param>
+        /// <param name="requestHeaders">The headers which will be included with the request. May be null.</param>
+        /// <param name="contentHeaders">The headers for the content which will be included with the request.
+        /// May be null.</param>
+        /// <remarks>
+        /// <para>
+        /// When either of the headers collections are specified, there are certain headers which will be included in the
+        /// signed URL's signature, and therefore must be included in requests made with the created URL. These are the
+        /// Content-MD5 and Content-Type content headers as well as any content or request header with a name starting
+        /// with "x-goog-". However, this does not apply to x-goog-encryption-key and x-goog-encryption-key-sha256 headers,
+        /// but they must be included with the request anyway if the desired content is encrypted with customer-supplied
+        /// encryption keys.
+        /// </para>
+        /// <para>
+        /// If the headers collections are null or empty, no headers are included in the signed URL's signature, so any
+        /// requests made with the created URL must not contain Content-MD5, Content-Type, or any header started with "x-goog-".
+        /// </para>
+        /// <para>
+        /// Note that when GET is specified as the <paramref name="requestMethod"/> (or it is null, in which case GET is
+        /// used), both GET and HEAD requests can be made with the created signed URL.
+        /// </para>
+        /// <para>
+        /// See https://cloud.google.com/storage/docs/access-control/signed-urls for more information on signed URLs.
+        /// </para>
+        /// </remarks>
+        /// <exception cref="InvalidOperationException">
+        /// The <paramref name="credentialFilePath"/> does not refer to a valid JSON service account key file.
+        /// </exception>
+        /// <returns>
+        /// The signed URL which can be used to provide access to a bucket or object for a limited amount of time.
+        /// </returns>
+        public static string Create(
+            string bucket,
+            string objectName,
+            string credentialFilePath,
+            DateTimeOffset? expiration,
+            HttpMethod requestMethod = null,
+            Dictionary<string, IEnumerable<string>> requestHeaders = null,
+            Dictionary<string, IEnumerable<string>> contentHeaders = null)
+        {
+            StorageClientImpl.ValidateBucketName(bucket);
+            GaxPreconditions.CheckNotNull(credentialFilePath, nameof(credentialFilePath));
+
+            var credentials = ParseCredentials(credentialFilePath);
+            var expires = ((int?)((expiration - UnixEpoch)?.TotalSeconds))?.ToString(CultureInfo.InvariantCulture);
```

Comment:
I'd call it `expiryUnixSeconds` - then I think it's really clear. (You know I'll always be picky with date/time stuff :)